doc/articles/howto-consume-webservices.md

```diff
@@ -0,0 +1,1849 @@
+
+    Then, looking at parameters, the **backingVariable** is passed by reference, allowing this method to change the value for the caller as well as locally. Next, the **value** is supplied. Finally, the **propertyName** parameter is defined. At first glance, this appears to be an optional parameter with a default value of null. However the compiler notes the presence of the **CallerMemberName** attribute and assigns the name of the calling method to the property (in the case of a property setter method, the name of the property).
+
+    The method body then compares the incoming value with the current property value, and returns false if they are equal. This prevents a situation where a property is updated, which updates a control via binding, which then sets the same property with the same value again. Without this guard, the code would continue to loop.
+
+    The **value** is then assigned to the **backingVariable** (updating the variable in the caller). The property changed event is raise (you will add that next) and a true value is returned.
+
+    > [!TIP]
+    > You can learn more about **CallerMemberNameAttribute** here - [Reserved attributes: Determine caller information](https://docs.microsoft.com/dotnet/csharp/language-reference/attributes/caller-information).
+
+1. To add the method that will actually raise the property changed event, locate the comment **// Insert RaisePropertyChanged below here** and replace it with the following code:
+
+    ```csharp
+    // Insert RaisePropertyChanged below here
+    protected void RaisePropertyChanged([CallerMemberName] string propertyName = null)
+    {
+    #pragma warning disable CS4014 // Because this call is not awaited, execution of the current method continues before the call is completed
+        DispatchAsync(() => PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName)));
+    #pragma warning restore CS4014 // Because this call is not awaited, execution of the current method continues before the call is completed
+    }
+    ```
+
+    This method uses a helper method (added next) to ensure the event is raised on the UI thread. As the code does not await this execution, the compiler would usually display a warning in the assumption that this is a mistake. As it is intentional, the code includes `#pragma` directives that temporarily disable that warning.
+
+    > [!TIP]
+    > If you want to learn more about lambda expressions, review the document here - [Lambda expressions](https://docs.microsoft.com/dotnet/csharp/language-reference/operators/lambda-expressions)
+
+1. To add a helper method that will ensure a callback is executed on the UI thread, locate the comment **// Insert Dispatch below here** and replace it with the following code:
+
+    ```csharp
+    // Insert DispatchAsync below here
+    protected async Task DispatchAsync(DispatchedHandler callback)
```

Comment:
OK - if @jeromelaban has input on this it is welcome. What would you guys recommend as a good practice here? Leave the current behavior and call out the idiosyncrasy? Detect __WASM__ and override the default behavior using the CoreDispatcher.HasThreadAccessOverride static flag? Some other option?

Reply:
@darenm The most common practice is probably to add WASM-specific code, eg:
```csharp
var hasThreadAccess =
#if __WASM__
true;
#else
Dispatcher.HasThreadAccess
#endif

This can be wrapped in a static helper class for reuse, but it's safer to do it locally to the app than to override the global `HasThreadAccessOverride` flag because that might break 3rd-party frameworks. (Except that some 3rd-party frameworks break if `HasThreadAccessOverride` is _not_ overridden...)

Reply:
@davidjohnoliver I have pushed an update to this and the associated sample.